client/my-sites/types/main.jsx

```diff
@@ -81,7 +81,7 @@ export default connect( ( state, ownProps ) => {
 	return {
 		siteId,
 		postType,
-		postTypeSupported: isPostTypeSupported( state, siteId, ownProps.query.type ),
-		userCanEdit: canCurrentUser( state, siteId, capability ),
+		postTypeSupported: isPostTypeSupported( state, siteId, ownProps.query.type ) || false,
```

Comment:
I wonder if we should have been more consistent in the return value of the selectors in the first place:
https://github.com/Automattic/wp-calypso/blob/b5b9732/client/state/post-types/selectors.js#L117

Reply:
yea, good point, thanks for investigating that a bit deeper! I'll look through that and try to get a more consistent solution

Reply:
I've updated it to change the selectors to return a true or false boolean, the canCurrentUser selector is used in quite a few places, I searched through them and actually it seems like the `null` return value was almost always just treated as `false`. This allowed me to clean up quite a few redundant type coercions.

Reply:
> The original use case for returning null would have been to prevent showing not authorized errors prematurely, and in fact, when loading the reusable blocks list page for the first time now, you do briefly see a permission error!

Thanks for changing these and explaining the reason why it was done in the first place. I didn't realize the checks were so pervasive.

I'm sorry to say – and it's my fault for not checking before I made the comment –  that maybe we shouldn't go so far in this PR since it's changing so many different areas.

Apologies for not mentioning this earlier and wasting your time 😞

I can revert back to https://github.com/Automattic/wp-calypso/pull/45346/commits/e33e7f21d0eee9d28ee8033c11ce8442428cd7cf and retest. The initial changes did look good to me, even from a defensive standpoint.

Reply:
Also, to elaborate - they are good changes, but probably better suited for another PR

Reply:
Actually, while I've been testing this diff, I checked all the pages and things just work, so maybe I was too conservative. 

So ignore my comments from above 🤣